src/buildstream/_remotespec.py

```diff
@@ -253,6 +266,10 @@ def new_from_node(
         if auth_node:
             server_cert, client_key, client_cert = cls._parse_auth(auth_node, basedir)
 
+        keepalive_config = spec_node.get_mapping("keepalive-config", None)
+        if keepalive_config is not None:
+            keepalive_config.validate_keys(["keepalive-time"])
```

Comment:
I think this is too much keepalive. We have only one keepalive setting, so no need to put it in a keepalive-config dictionary. We should either have it directly in the server config or (and I think this would be better) in a generic dictionary for connection settings that allows adding things later.

Reply:
This was suggestion by @juergbi if I understood right. There are about ten keepalive settings.

Reply:
This is anyway reasonably localized thing to change. The reasoning here was that full grpc keepalive spec looks like this https://grpc.io/docs/guides/keepalive/#keepalive-configuration-specification

Reply:
I converted this to more generic connection-config.